src/zeit/content/cp/blocks/automatic.py

```diff
@@ -87,6 +87,24 @@ def layout(self, layout):
     def change_layout(self, layout):
         self.temporary_layout = layout
 
+    def materialize(self):
+        # Most of this code is quite similar to Area._materialize_auto_blocks,
+        # but not similar enough to be extracted.
+        area = self.__parent__
+
+        block_id = self.__name__
+        rendered = zeit.content.cp.interfaces.IRenderedArea(area)
+        for block in rendered.values():
+            if block.__name__ == block_id:
+                auto_filled = block
+                break
+
+        order = area.keys()
+        del area[block_id]
+        materialized = area.create_item('teaser')
+        area.copy_teaserlist_attributes(auto_filled, materialized)
+        area.updateOrder(order)
```

Comment:
Das Schema `order = area.keys()` und `area.updateOrder(order)` verwenden wir mittlerweile an einigen Stellen. Ergibt es Sinn das ganze etwas besser zu beschreiben, indem es einen Context Manager auf `Area` gibt, damit man z.B. schreiben kann `with area.keep_order(): ...`? Dann sieht man durch die Einrückung welcher Teil beeinflusst wird und der Code spricht besser für sich selbst.

Ich kann mir auch gut vorstellen, dass man dann `keep_order` erweitert, sodass es ggf. sogar möglich ist, die Reihenfolge der Blöcke beizubehalten, wenn man weitere hinzufügt. Dann wären die alten Blöcke (von `keys = area.keys()`) am Anfang in alter Reihenfolge und die neuen Blöcke sind unten in der Reihenfolge, in der man sie vorfindet. Auch Handling von gelöschten Blöcken könnte man dann einbauen, weil es eine zentrale Stelle dafür gibt.

Also ein Context Manager damit folgendes funktioniert:

    with area.keep_order():
        area.create_item('teaser')
        del area[area.keys()[0]]

Zusätzliche und gelöschte Blöcke hatten mir beim umsetzen nämlich ein paar Probleme gemacht und ich wäre durchaus froh, wenn sich der Context Manager darum kümmern könnte. (Man kann ja vielleicht per Argument aussteuern, ob es erlaubt ist dass die Keys sich ändern.)